ash/src/vk.rs

```diff
@@ -82,6 +66,103 @@ pub(crate) unsafe fn ptr_chain_iter<T: ?Sized>(
     })
 }
 
+/// Structures implementing this trait are layout-compatible with [`BaseInStructure`] and
+/// [`BaseOutStructure`]. Such structures have an `s_type` field indicating its type, which must
+/// always match the value of [`TaggedStructure::STRUCTURE_TYPE`].
+pub unsafe trait AnyTaggedStructure<'a> {
+    /// Prepends the given extension struct between the root and the first pointer. This
+    /// method only exists on structs that can be passed to a function directly. Only
```

Comment:
> This method only exists on structs that can be passed to a function directly

Is that still true?

Reply:
If I'm not mistaken, there exists no `Extends<T>` (based on `vk.xml` `structextends="T"`) for a `T` that is not a "root structure" (~= structure that can be passed to a function). Hence there's no way to satisfy the `T: Extends<Self>` bound (enabling this `fn push()` function) unless `Self` was such a root structure?

Reply:
Ah, right. I think "exists" is slightly confusing word choice for this, since the method does literally exist on "any tagged structure", but for non-root structures there are not (normally) any legal *arguments* to it. Maybe:
```suggestion
    /// method is only useful on structs that can be passed to a function directly. Only
```

Reply:
Perhaps the wording `is only available` solves that? Perhaps `available` carries the right weight of "it exists, but you may not always be able to use it".